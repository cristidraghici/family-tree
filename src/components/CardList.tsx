const CardList = () => {
  return <>Cards not implemented yet</>
}
export default CardList
