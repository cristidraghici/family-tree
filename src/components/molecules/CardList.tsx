import { FunctionComponent, Fragment } from 'react'
import ConditionalElement from '../atoms/ConditionalElement'
import { ExtendedPersonType } from '@/utils/PersonRegistryUtil'

const CardList: FunctionComponent<{ persons: ExtendedPersonType[] }> = ({ persons }) => {
  return (
    <>
      <ConditionalElement as="div" className="CardsGrid" condition={persons.length > 0}>
        {persons.map((person: ExtendedPersonType) => (
          <article key={person.id}>
            <header>{person.fullName}</header>
            {person.biography}
            <footer>
              <ConditionalElement as="p" condition={!!person.parentsNames}>
                <strong>Parents</strong>: {person.parentsNames}
              </ConditionalElement>
              <ConditionalElement as="p" condition={!!person.relativesNames}>
                <strong>Relatives</strong>: {person.relativesNames}
              </ConditionalElement>
            </footer>
          </article>
        ))}
      </ConditionalElement>

      <ConditionalElement as={Fragment} condition={persons.length === 0}>
        <article>No persons found.</article>
      </ConditionalElement>
    </>
  )
}
export default CardList
