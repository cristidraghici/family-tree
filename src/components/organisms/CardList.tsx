import { FunctionComponent, Fragment } from 'react'

import ConditionalElement from '@/components/atoms/ConditionalElement'
import Card from '@/components/molecules/Card'

import { ExtendedPersonType } from '@/utils/PersonRegistryUtil'

const CardList: FunctionComponent<{
  persons: ExtendedPersonType[]
  onEdit: (person: ExtendedPersonType) => void
}> = ({ persons, onEdit }) => {
  return (
    <>
      <ConditionalElement as="div" className="CardsGrid" condition={persons.length > 0}>
        {persons.map((person: ExtendedPersonType) => (
          <Card key={person.id} person={person} onClick={() => onEdit(person)} />
        ))}
      </ConditionalElement>

      <ConditionalElement as={Fragment} condition={persons.length === 0}>
        <article>No persons found.</article>
      </ConditionalElement>
    </>
  )
}
export default CardList
