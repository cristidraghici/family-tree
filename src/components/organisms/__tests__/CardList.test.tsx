import { render } from '@testing-library/react'
import CardList from '../CardList'

describe('CardList', () => {
  it('renders correctly', () => {
    const { container } = render(<CardList />)

    expect(container).toMatchSnapshot()
  })
})
