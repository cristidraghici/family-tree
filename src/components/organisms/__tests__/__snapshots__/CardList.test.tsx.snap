// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CardList > renders correctly 1`] = `
<div>
  <article>
    No persons found.
  </article>
</div>
`;
